Improve the way how log part between timestamps is retrieved using awk

pmihalcin opened the pull request:
Take into consideration logs where each log line doesn't start with timestamp

https://github.com/EmbedITCZ/jbehave-support/issues/8

thradec commented on jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshTemplate.java:
```diff
@@ -1,109 +1,109 @@
+    public SshTemplate(SshSetting sshSetting,
+                       String timestampFormat, String timestampPosixRegex,
+                       RollingLogResolver rollingLogResolver) {
+        isTrue(!isEmpty(sshSetting.getLogPath()), "log path must not be null or empty");
```
I think this is backward incompatible change, I'd keep the original version with 3 parameters and add the new one

pmihalcin replied:
I pushed new solution which doesn't rely on timestampRegex, hence is backward compatible.

thradec commented:
I think we should recreate the original jira issue in github issue tracker and avoid using JBUS-xy

thradec commented:
minor: your commit use HCG user, I'm not sure if it intentional

paveljandejsek commented on jbehave-support-core/src/main/resources/get-log-between-timestamps.awk:
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/awk -f
```
Is this file used?

paveljandejsek commented on jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshTemplate.java:
```diff
@@ -1,109 +1,109 @@
+                                     final String timestampPosixRegex,
+                                     final String logPath) throws IOException {
+        DateTimeFormatter formatter = ofPattern(timestampFormat);
+        String command = copyToString(new ClassPathResource("get-log-between-timestamps-template.awk").getInputStream(),
```
Might be better to move it to class level so the file wont get loaded each time the log copying is called.

pmihalcin replied:
I addressed it. Thanks for the suggestion.

paveljandejsek commented:
Do we think there might be a solution to this issue by leveraging the existing timestampFormat without introducing the additional timestampPosixRegex? (not saying there is, just putting it up to try to avoid adding new config parameter if we can *somehow* avoid it :slightly_smiling_face: )

pmihalcin commented:
> I think we should recreate the original jira issue in github issue tracker and avoid using JBUS-xy

Done: https://github.com/EmbedITCZ/jbehave-support/issues/8

pmihalcin commented:
> minor: your commit use HCG user, I'm not sure if it intentional

I changed it. Thanks

pmihalcin commented:
> Do we think there might be a solution to this issue by leveraging the existing timestampFormat without introducing the additional timestampPosixRegex? (not saying there is, just putting it up to try to avoid adding new config parameter if we can _somehow_ avoid it 🙂 )

I was thinking about it, but I couldn't figure out how to do it. If anybody knows, let me know and I'll incorporate it.

paveljandejsek commented:
> > Do we think there might be a solution to this issue by leveraging the existing timestampFormat without introducing the additional timestampPosixRegex? (not saying there is, just putting it up to try to avoid adding new config parameter if we can _somehow_ avoid it slightly_smiling_face )
> 
> I was thinking about it, but I couldn't figure out how to do it. If anybody knows, let me know and I'll incorporate it.

I think I found a solution without introducing the regex, give me a couple of days to verify it - I will post an update once I have something working (or find out it cant be done)

pmihalcin commented:
@thradec @paveljandejsek 

I pushed alternative solution which doesn't break backward compatibility.
I also addressed all comments.
Have a look again and let me know if there is something to add or merge.

Thanks

paveljandejsek commented on jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshTemplate.java:
```diff
@@ -51,59 +57,43 @@ public SshSetting getSshSetting() {
-        return logContents;
+        DateTimeFormatter formatter = ofPattern(timestampFormat);
+        String getLogBetweenTimestamps = format(getLogBetweenTimestampsCommand,
+            startTime.format(formatter), endTime.format(formatter), endTime.plusYears(1).format(formatter), logPath);
```
I would also add one minute and one second just to be safe in case that the user does not use timestamp with year and just uses minutes and seconds or just seconds like in unix timestamp.

```suggestion
            ZonedDateTime upperLimitTimestamp = endTime.plusYears(1).plusMinutes(1).plusSeconds(1);
            startTime.format(formatter), endTime.format(formatter), upperLimitTimestamp.format(formatter), logPath);
```

paveljandejsek commented on jbehave-support-core/src/main/resources/get-log-between-timestamps-template.awk:
```diff
@@ -0,0 +1,57 @@
+    # use different pointer to inputFile than the current one
+    getline line < inputFile;
+  }
+  print line;
```
Just a question - wouldn't the following:
```
for (i = 0; i <= endLineNumber - startLineNumber; i++) {
    print line;
    getline line < inputFile;
  }
```
get rid of the `print line;` between the two for cycles? Or is it a bit more complicated and we cant get rid of `print line;` twice so easily 🙂
